core/src/main/scala/org/apache/spark/ui/ConsoleProgressBar.scala

```diff
@@ -121,5 +121,8 @@ private[spark] class ConsoleProgressBar(sc: SparkContext) extends Logging {
    * Tear down the timer thread.  The timer thread is a GC root, and it retains the entire
    * SparkContext if it's not terminated.
    */
-  def stop(): Unit = ThreadUtils.shutdown(timer)
+  def stop(): Unit = {
+    timerFuture.cancel(true)
+    ThreadUtils.shutdown(timer)
+  }
```

Comment:
Thanks for the change @beliefer . 
Interested in knowing how this change in the ui consoleprogress bar was tested?

Reply:
Good question!

Reply:
To be honestly, I don't know it too.